pkg/controller/statefulset/stateful_set_control.go

```diff
@@ -622,11 +623,36 @@ func (ssc *defaultStatefulSetControl) rollingUpdateStatefulsetPods(
 		}
 	}
 
+	// handle specified deleted pod, and protected by maxUnavailable
```

Comment:
// handle specified deleted pod under maxUnavailable constrain
// NOTE: specified deletion is not constraint by partition setting